core/src/main/scala/org/apache/spark/network/netty/SparkTransportConf.scala

```diff
@@ -44,8 +43,11 @@ object SparkTransportConf {
     // assuming we have all the machine's cores).
     // NB: Only set if serverThreads/clientThreads not already set.
     val numThreads = NettyUtils.defaultNumThreads(numUsableCores)
-    conf.setIfMissing(s"spark.$module.io.serverThreads", numThreads.toString)
-    conf.setIfMissing(s"spark.$module.io.clientThreads", numThreads.toString)
+    val finalServerNumThreads = getNumOfThreads(conf, module, true, numThreads)
+    val finalClientNumThreads = getNumOfThreads(conf, module, false, numThreads)
+
+    conf.setIfMissing(s"spark.$module.io.serverThreads", finalServerNumThreads)
+    conf.setIfMissing(s"spark.$module.io.clientThreads", finalClientNumThreads)
```

Comment:
We should actually set spark.driver.xxx, spark.executor.xxx here?

Reply:
I see. We have different TransportConf for driver and executor.

Reply:
yes. your are correct.